lib/api/href.rb

```diff
@@ -1,34 +1,57 @@
 module Api
-  class HrefParser
-    def self.parse(href)
-      new(href).parse
-    end
-
+  class Href
     def initialize(href)
       @href = href
     end
 
-    def parse
-      return [nil, nil] unless href
-      if subcollection?
-        [subcollection.to_sym, ApplicationRecord.uncompress_id(subcollection_id)]
-      else
-        [collection.to_sym, ApplicationRecord.uncompress_id(collection_id)]
-      end
+    def resource
+      (subcollection? ? subcollection : collection)&.to_sym
```

Comment:
Do you actually need the call to `to_proc` (`&`) here to make this work? Doesn't (subcollection? ? subcollection : collection).to_sym` do the same thing?

Reply:
Heh, and I just noticed that Rubocop is exploding with a syntax error after I stared at this for a bit wondering if the omission of `:` here is intended. Looks like it's not ;) Indeed I think you can just do away with the ampersand.

Reply:
I needed this in the case that the expression in parentheses evaluates to `nil` : https://github.com/ManageIQ/manageiq-api/pull/97/commits/e88cef9e103d4ed17b5289160dbfb3a0159290fe

*Having said that*, I'm kind of questioning why this needs to be a symbol....I've mostly tried to preserve the API from what this started as. We will either be interpolating it or using it to lookup some configuration. In the latter case, if we don't do this already, it should be flexible enough for us to provide either a symbol/string. I'll look into removing the `to_sym`